great_expectations/expectations/metrics/util.py

```diff
@@ -1058,3 +1059,83 @@ def get_sqlalchemy_source_table_and_schema(
         )
     else:
         return engine.batch_manager.active_batch_data.selectable
+
+
+def get_unexpected_indices_for_multiple_pandas_named_indices(
+    df: pd.DataFrame,
+    expectation_domain_column_name: str,
+    unexpected_index_column_names: List[str],
+) -> List[Dict[str, Any]]:
+    """
+    Builds unexpected_index list for Pandas Dataframe in situation where the named
+    columns is also a named index. This method handles the case when there are multiple named indices.
+    Args:
+        df: reference to Pandas dataframe
+        expectation_domain_column_name: column that Expectation is being run for
+        unexpected_index_column_names: column_names for indices, either named index or unexpected_index_columns
+
+    Returns:
+        List of Dicts that contain ID/PK values
+    """
+    if not expectation_domain_column_name:
+        raise ge_exceptions.MetricResolutionError(
+            message=f"Error: The domain column is currently set to None. Please check your configuration.",
```

Comment:
We could get this error if `expectation_domain_column_name` is the empty string `""` so we might want to determine if it's `None` or `""` and update the error message. If this is only called by our code (vs a user facing api) and we are mypy checking this should never be `None`. However, I know we do not have everything covered by mypy.

Reply:
`if expectation_domain_column_name is None` and make `Optional` above or `Union[str, None]`

Reply:
This is now done as well